### PR TITLE
impr: only display required confidential info FPLA-1175

### PR DESCRIPTION
**Pull request by swalker125**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FPLA-1175

### Change description ###

Only display:

-     first name
-     last name
-     address
-     email
-     phone number

For respondent and child data strucutres in confidential tabs

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by swalker125** [comment]
Screenshot_2020-03-25 Core Case Data HMCTS
Screenshot_2020-03-25 Core Case Data HMCTS(1)

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/ChildrenService.java [hunk 1]
```diff
@@ -23,15 +23,22 @@
         List<Element<Child>> childCollection = new ArrayList<>();
 
         if (caseData.getAllChildren().isEmpty()) {
-            childCollection.add(emptyElementWithPartyId());
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/ChildrenService.java]: Any particular reason for inlining this?

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/RespondentService.java [hunk 2]
```diff
@@ -23,15 +23,25 @@
         List<Element<Respondent>> respondentsCollection = new ArrayList<>();
 
         if (caseData.getAllRespondents().isEmpty()) {
-            respondentsCollection.add(emptyElementWithPartyId());
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/RespondentService.java]: see previous

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/ChildrenService.java [hunk 3]
```diff
@@ -40,11 +47,16 @@
         return childCollection;
     }
 
-    // expands collection in UI. A value (in this case partyId) needs to be set to expand the collection.
-    private Element<Child> emptyElementWithPartyId() {
-        return ElementUtils.element(Child.builder()
-            .party(ChildParty.builder().partyId(randomUUID().toString()).build())
-            .build());
+    private Child addConfidentialDetails(Element<Child> confidentialChild, Element<Child> child) {
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/ChildrenService.java]: only need to send the party part of `confidentialChild`

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/ChildrenService.java [hunk 4]
```diff
@@ -73,14 +85,28 @@
...
+        List<Element<Child>> confidentialChildrenModified = new ArrayList<>();
+
+        confidentialChildren.forEach(element -> confidentialChildrenModified.add(
+            element(element.getId(), Child.builder().party(
+                ChildParty.builder()
+                    .firstName(element.getValue().getParty().firstName)
+                    .lastName(element.getValue().getParty().lastName)
+                    .address(element.getValue().getParty().address)
+                    .telephoneNumber(element.getValue().getParty().telephoneNumber)
+                    .email(element.getValue().getParty().email)
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/ChildrenService.java]: could extract the party to a variable before the add if you wanted

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/RespondentService.java [hunk 5]
```diff
@@ -40,11 +50,17 @@
         return respondentsCollection;
     }
 
-    // expands collection in UI. A value (in this case partyId) needs to be set to expand the collection.
-    private Element<Respondent> emptyElementWithPartyId() {
-        return ElementUtils.element(Respondent.builder()
-            .party(RespondentParty.builder().partyId(randomUUID().toString()).build())
-            .build());
+    private Respondent addConfidentialDetails(Element<Respondent> confidentialRespondent,
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/RespondentService.java]: only need to pass the party part of `confidentialRespondent`

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/RespondentService.java [hunk 6]
```diff
@@ -82,6 +98,23 @@
...
+        List<Element<Respondent>> confidentialRespondentsModified = new ArrayList<>();
+
+        confidentialRespondents.forEach(element -> confidentialRespondentsModified.add(
+            element(element.getId(), Respondent.builder().party(
+                RespondentParty.builder()
+                    .firstName(element.getValue().getParty().firstName)
+                    .lastName(element.getValue().getParty().lastName)
+                    .address(element.getValue().getParty().address)
+                    .telephoneNumber(element.getValue().getParty().telephoneNumber)
+                    .email(element.getValue().getParty().email)
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/RespondentService.java]: could extract the party to a variable before the add if you wanted

**Pull request closed without merging**